### PR TITLE
refactor: dialog for new listing, icon for complete column

### PR DESCRIPTION
**Pull request by bigopon**
Summary

  * new `jsconfig.json` to help resolve path for intellisense
  * new `util-components` folder to hold business-logic irrelevant components
  * new `dlg` folder to hold all dialog components
  * `listing-list` changes to use dialog instead of inline editor (or inline-editor is more desired ?)
  * some `jsdoc` annotations for intellisense all over the places 😄

### CODE REVIEW

File: client/src/components/dlg/new-listing.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<template>
+  <ux-dialog class='new-listing-dlg'>
```
Comment (plwalters) [inline client/src/components/dlg/new-listing.html]: @Vheissu put together some modal styles that are pretty nice, can we switch these those styles?  It should be reflected in the other modal code (ex here - https://github.com/AureliaToolbox/gigme.io-client/blob/master/client/src/components/show-all-addresses.html)

File: client/src/components/dlg/new-listing.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <ux-dialog-header>
+      <h2>New Listing</h2>
+      <div>Comapny: <b>${newListing.company.name}</b></div>
```
Comment (plwalters) [inline client/src/components/dlg/new-listing.html]: Typo

File: client/src/components/dlg/new-listing.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    </ux-dialog-header>
+
+    <ux-dialog-body>
+      <label>
+        Title
+        <input value.bind='newListing.title' />
```
Comment (plwalters) [inline client/src/components/dlg/new-listing.html]: Prefer `"` over `'` in html.

**Comment by bigopon** [comment]
@PWKad done 😄 

I was wondering if we have a dialog style predefined too

**Review by plwalters — APPROVED** [review]

**Pull request merged**